sysapi/sysapi_util/Unmarshal_TPML_INTEL_PTT_PROPERTY.c

```diff
@@ -0,0 +1,50 @@
+/***********************************************************************;
+* Copyright (c) 2017, Intel Corporation
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice,
+* this list of conditions and the following disclaimer.
+*
+* 2. Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following disclaimer in the documentation
+* and/or other materials provided with the distribution.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+* THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************/
+
+#include <sapi/tpm20.h>
+#include "sysapi_util.h"
+
+void Unmarshal_TPML_INTEL_PTT_PROPERTY(
```

Comment:
Ideally we'd be able to use the same function for other TPM vendor capabilities but the spec doesn't define the type so this one is specific to Intel's PTT. It's very likely that a different TPM would use a different type, likely a tagged property unlike the raw UINT32 used by PTT.

So you're right: this commit only adds support for PTT and it probably won't work with other vendor specific properties. I figure we can make this more dynamic in the future if we get any requests for support from other vendors. No need to make it too complicated in the first iteration.